sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -117,12 +198,14 @@ static pi_result getExtFuncFromContext(pi_context context, T *fptr) {
 
   if (!FuncPtr) {
     // Cache that the extension is not available
-    FuncPtrs[context] = nullptr;
+    FuncInitialized.first = nullptr;
+    FuncInitialized.second = true;
```

Comment:
Both paths set this to true. Would it make sense to move it to before the if-statement and have a comment stating the intent of setting it to `true`?

Reply:
Done